setup/includes/upgrades/common/3.0.0-media-soure-type-system-setting.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Add new system setting for the default media source type.
+ *
+ * @var modX
+ */
+
+$messageTemplate = '<p class="%s">%s</p>';
+
+$defaultMediaSourceTypeSystemSetting = $modx->getObject('modSystemSetting', array(
+    'key' => 'default_media_source_type',
+));
+if ($defaultMediaSourceTypeSystemSetting === null) {
+    $defaultMediaSourceTypeSystemSetting = $modx->newObject('modSystemSetting');
+    $defaultMediaSourceTypeSystemSetting->set('key','default_media_source_type');
```

Comment:
This code can be simplified using `fromArray()` method. It will become more readable.